### DISCUSSION TITLE
nginx-proxy not working with docker on rootless mode due to lack of permissions

### DISCUSSION THREAD

**Discussion by lucasrla** [nginx-proxy/nginx-proxy#1683] [Q&A]
Hello, buchdag. Thank you so much for maintaining this great piece of software. It's been extremely useful to me.

I just wanted to point out that I could not get nginx-proxy to work with [docker on rootless mode](https://docs.docker.com/engine/security/rootless/). Even after [exposing the privileged ports (below 1204)](https://docs.docker.com/engine/security/rootless/#exposing-privileged-ports).

The error I got was exactly the [one mentioned by you in this comment](https://github.com/nginx-proxy/acme-companion/issues/706#issuecomment-721301768):

> dockergen encounter an issue inside the nginx-proxy container (seems like it can't access the Docker socket because of pemission issues):
>
> ```
> dockergen.1 | 2020/11/03 11:00:21 Error retrieving docker server version info: Get http://unix.sock/version: dial unix /tmp/docker.sock: connect: permission denied
> dockergen.1 | 2020/11/03 11:00:21 Error listing containers: Get http://unix.sock/containers/json?all=1: dial unix /tmp/docker.sock: connect: permission denied
> ```

The only way to make it work (that I could find) was to uninstall the rootless mode and go back to `sudo`ing.

Do you have any tips or suggestions about how to make nginx-proxy work within docker rootless mode?

***

For the record, what have I tried:

- Running `docker` and `docker-compose` with `sudo`
- Adding `privileged: true` to services in docker-compose YAML file
- Adding my user to the `docker` group

None of these things worked for me.

Related StackOverflow (or else) threads, also for the record:

- [“dial unix /tmp/docker.sock: connect: permission denied” when running docker-compose up](https://stackoverflow.com/questions/58390960/dial-unix-tmp-docker-sock-connect-permission-denied-when-running-docker-com)
- [Docker mounted volume GID mismatch causes error: dial unix /var/run/docker.sock: connect: permission denied](https://stackoverflow.com/questions/46632829/docker-mounted-volume-gid-mismatch-causes-error-dial-unix-var-run-docker-sock)
- [Docker error with /var/run/docker.sock](https://discussion.fedoraproject.org/t/docker-error-with-var-run-docker-sock/2425)

**Comment by snappermorgan** [comment]
I found the answer to this scenario here on stack overflow. https://stackoverflow.com/questions/68331153/docker-rootless-error-you-need-to-share-your-docker-host-socket-with-a-volume-a

Basically, use the docker.sock file in the path specified by XDG_RUNTIME_DIRE 

echo $XDG_RUNTIME_DIR

For me, I mounted the volume in my docker-compose.yml file with:

/run/user/1000/docker.sock:/tmp/docker.sock:ro

Finally got it to work.
